trustmanager/keydbstore.go

```diff
@@ -0,0 +1,153 @@
+package trustmanager
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/docker/notary/pkg/passphrase"
+	"github.com/endophage/gotuf/data"
+	"github.com/jinzhu/gorm"
+	gojose "github.com/square/go-jose"
+)
+
+// KeyDBStore persists and manages private keys on a SQL database
+type KeyDBStore struct {
+	sync.Mutex
+	db         gorm.DB
+	passphrase string
+	encrypter  gojose.Encrypter
+	cachedKeys map[string]data.PrivateKey
+}
+
+// GormPrivateKey represents a PrivateKey in the database
+type GormPrivateKey struct {
+	gorm.Model
+	KeyID      string `sql:"not null;unique"`
+	Encryption string `sql:"not null"`
+	Algorithm  string `sql:"not null"`
+	Public     []byte `sql:"not null"`
```

Comment:
An artifact from gojose receiving a string for encryption and decryption (I think it should receive []byte).

The field that doesn't need to be encrypted/decrypted kept the right data-type :)

I'll change Private to byte[] and convert it before encryption/after decryption.